index.js

```diff
@@ -583,6 +626,11 @@
     return typeof other === typeof this && other.valueOf() === this.valueOf();
   }
 
+  //  String$prototype$lte :: String ~> String -> Boolean
+  function String$prototype$lte(other) {
+    return typeof other === typeof this && this.valueOf() <= other.valueOf();
+  }
```

Comment:
If we decide to continue to treat `'abc'` and `new String('abc')` as different values, this `typeof` check will remain necessary. At the top level we ensure that `type(x) === type(y)`, but `type` doesn't differentiate between string primitives and String objects.